edx_notifications/stores/sql/store_provider.py

```diff
@@ -0,0 +1,283 @@
+"""
+Concrete MySQL implementation of the data provider interface
+"""
+
+import copy
+import pylru
+
+from django.core.exceptions import ObjectDoesNotExist
+
+from edx_notifications.stores.store import BaseNotificationStoreProvider
+from edx_notifications.exceptions import (
+    ItemNotFoundError,
+    ItemIntegrityError
+)
+from edx_notifications import const
+from edx_notifications.stores.sql.models import (
+    SQLNotificationMessage,
+    SQLNotificationType,
+    SQLNotificationUserMap
+)
+
+
+class SQLNotificationStoreProvider(BaseNotificationStoreProvider):
+    """
+    Concrete MySQL implementation of the abstract base class (interface)
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializer
+        """
+
+        _msg_type_cache_size = kwargs.get('MAX_MSG_TYPE_CACHE_SIZE', 1024)
+        self._msg_type_cache = pylru.lrucache(_msg_type_cache_size)
+
+    def _get_notification_by_id(self, msg_id, options=None):
+        """
+        Helper method to get Notification Message by id
+        """
+
+        # pylint/pep8 seem to complain if defaults are set to empty dicts
+        _options = options if options else {}
+        select_related = _options.get('select_related', True)
+
+        try:
+            if select_related:
+                obj = SQLNotificationMessage.objects.select_related().get(id=msg_id)
+            else:
+                obj = SQLNotificationMessage.objects.get(id=msg_id)
```

Comment:
Actually, I guess it would need to be:


    query = SQLNotificationMessage.objects
    if select_related:
        query = query.select_related()
    obj = query.get(id=msg_id)